crates/testing/src/overall_safety_task.rs

```diff
@@ -195,6 +210,16 @@ impl<TYPES: NodeType, I: TestableNodeImplementation<TYPES>> TestTaskState
                 self.error = Some(Box::new(OverallSafetyTaskErr::<TYPES>::TooManyFailures {
                     failed_views: self.ctx.failed_views.clone(),
                 }));
+            } else if !expected_views_to_fail.is_empty()
+                && !expected_views_to_fail.contains(&view_number)
+            {
+                let _ = self.test_sender.broadcast(TestEvent::Shutdown).await;
```

Comment:
Looks like this exact block is duplicated. Could we perhaps make it a single helper or something? It lowers the refactor burden as someone could miss one of the copies.

Reply:
refactored